test/runtime/document_test.cc

```diff
@@ -72,19 +72,6 @@ describe("Document", [&]() {
         "(array (true) (false))");
     });
 
-    it("handles truncated UTF16 data", [&]() {
-      char *content = reinterpret_cast<char*>(malloc(1));
-
-      spy_input->content = string((const char *)content, 1);
-      spy_input->encoding = TSInputEncodingUTF16;
-
-      ts_document_set_input(document, spy_input->input());
-      ts_document_invalidate(document);
-      ts_document_parse(document);
-
-      free(content);
-    });
-
```

Comment:
I think there's some value in testing it through `ts_document_parse`. I'm guessing this test didn't trigger the OOB read because `std::string` appends a null byte? Maybe you could use `ts_document_set_input_string_with_length` here, with a manually-malloc'd  buffer (contrary to what I suggested earlier).